### ISSUE TITLE
stop locations placed directly on the tracks don't match the inputs

### ISSUE THREAD

**Issue by eckter** [OpenRailAssociation/osrd#9228]
### What happened?

When placing stops just before signals, the output actually moves stops around. The stops are not placed before the signals when running the simulations (according to "VISA" speed limits)

Path that's being displayed after the pathfinding request: 

![image](https://github.com/user-attachments/assets/9ca05e87-4f96-44e4-a821-07c0ebed091a)

Path that's being displayed when I click on "edit this train":

![image](https://github.com/user-attachments/assets/fb0d4bfd-84cd-4027-b9cd-1528fd2bd7b5)

Notice that the middle stop has moved. The second version seems to be used for the simulation, despite being different from what you'd expect (i.e. it's not just the prefilled path in the path edition that's wrong)

### What did you expect to happen?

The stop shouldn't move like that. Requesting a stop before a signal should make the train stop before a signal. 

### How can we reproduce it (as minimally and precisely as possible)?

1. Turn on the signal layer
2. Find a path at the track level, with one intermediate stop right before a signal
3. Add a train
4. Edit this new train, notice that the stop has moved

### On which environments the bug occurs?

Local

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

e0152b07c7f42535ec9abd363fce3e89399f2b38

**Comment by eckter** [comment]
After some investigation:

* It only happens when the track geography doesn't match the track lengths. This makes it a non-issue on real infra, it's only relevant in debug scenarios. I'm lowering the severity.
* The payloads seem to not have any issue, meaning that the bug is located in the front-end.

**Comment by eckter** [comment]
*Actually* it's not just "front-end is broken", it's an issue with how our APIs aren't robust to geo data anomalies.

The front-end only receives geo data over the whole path. When track lengths don't match their geo lengths, it's impossible to mitigate any kind of error. They'll inevitably accumulate over the whole path and we can't normalize them.

**Comment by bougue-pe** [comment]
This bug is ~probably~ fixed by #10515 (~to be verified:~ ✅ verified), but all (too much) in front.

New plan after discussion in https://github.com/OpenRailAssociation/osrd/issues/9372#issuecomment-2765815821:
1. core provides `geom_projection` in response of `/path_properties`
2. then in front, replace the work done in #10515 by providing and using the converter based on `geom_projection`
3. (and only after fixing this bug, fix #9110)

---
Deprecated plan:
> After first discussions in https://github.com/OpenRailAssociation/osrd/issues/9372#issuecomment-2639484994, the previous plan was to change the fix by:
> * following the TLDR (add geographical ratio to the topological offset in the response)
> * implem editoast's side first for `/path_properties` (and **NOT** /pathfinding) https://github.com/OpenRailAssociation/osrd/issues/11191
> * then in front, replace the work done in #10515 
> * (and only after fixing this bug, fix #9110)

**Comment by SharglutDev** [comment]
> This bug is ~probably~ fixed by [#10515](https://github.com/OpenRailAssociation/osrd/pull/10515) (~to be verified:~ ✅ verified), but all (too much) in front.
> 
> New plan after discussion in [#9372 (comment)](https://github.com/OpenRailAssociation/osrd/issues/9372#issuecomment-2765815821):
> 
>     1. core provides `geom_projection` in response of `/path_properties`

@bougue-pe Where are with with this ticket ? Don't think 1. has been done ? Is `geom_projection` different than `geometry` in path properties response ?

To display incompatible constraints we use what is returned by pathfinding_blocks when error type is `incompatible_constraints`.

Not sure to understand if the issue is the data returned by this endpoint or the geometry returned by path properties ?

**Comment by bougue-pe** [comment]
`geometry` is the shape of the path (with a corresponding geographic length). This has actually no relation with the topological length of the path, as it's the same for the track-sections used by the path (because track-section's geographical lengths are usually a little different than their topological length).
And all the track locations of the response are actually "speaking" in topological distance, not geographical distance.
So using those topological distances to display on a map is most of the time off by a few (can be a lot on a long path).

`geom_projection` (under paused development) aims to provide an "interpolation curve" of lengths all along the path (see https://github.com/OpenRailAssociation/osrd/issues/9372#issuecomment-2765815821)
And it's not done, it's been paused recently, there are some complications (tracked into https://github.com/OpenRailAssociation/osrd/issues/11191).
